nncf/quantization/algorithms/channel_alignment/algorithm.py

```diff
@@ -0,0 +1,475 @@
+from nncf.common.factory import ModelTransformerFactory
+from nncf.common.factory import NNCFGraphFactory
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.patterns import GraphPattern
+from nncf.common.graph.transformations.commands import TargetPoint
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.common.tensor_statistics.statistic_point import StatisticPoint
+from nncf.common.tensor_statistics.statistic_point import StatisticPointsContainer
+from nncf.common.utils.backend import BackendType
+from nncf.common.utils.backend import get_backend
+from nncf.quantization.algorithms.algorithm import Algorithm
+from nncf.quantization.algorithms.channel_alignment.backend import ALGO_BACKENDS
+from nncf.quantization.algorithms.channel_alignment.backend import ChannelAlignmentAlgoBackend
+from nncf.quantization.algorithms.channel_alignment.backend import LayoutDescriptor
+
+TModel = TypeVar("TModel")
+
+
+class ChannelAlignment(Algorithm):
+    """
+    Post-training ChannelAlignment algorithm implementation.
+
+    The main purpose of this algorithm to reduce quantization error
+    via correction the parameters of the Convolutions, FullyConnected and their biases.
+    Algorithm consists of following steps:
+        - algorithm is searching for convolution -> convolution pairs in the target model.
+        - minimal and maximal activations quantiles of first convolutions are collected on the target subset.
+        - algorithm calculates median of collected values, it is used then to adjust
+            convolution layers biases and weights.
+        - biases of matched subgraphs convolutions are adjusted, so mean points of first
+            convolution activations quantile medians are translated to zero.
+        - weights of matched subgraph convolutions are adjusted, so all first convolutions activations
+            which were between median of low quantile and median of high quantile are translated to [-1, 1] range.
+    """
+
+    def __init__(
+        self,
+        subset_size: int = 100,
+        inplace_statistics: bool = True,
+        backend_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        :param subset_size: Size of a subset for the statistics collection,
+            defaults to 100.
+        :param inplace_statistics: Defines wheather to calculate quantizers statistics
+            by backend graph operations or by default Python implementation, defaults
+            to True.
+        :param backend_params: Backend specific parameters.
+        """
+        super().__init__()
+        self.subset_size = subset_size
+        self.inplace_statistics = inplace_statistics
+        self.backend_params = backend_params
+        self._original_nncf_graph = None
+        self._backend_entity = None
+        self._quantile = 1e-4
```

Comment:
Why this param can not be changed?

Reply:
I copied POT logic, in POT it's hardcoded. Do you think we should expose this? Let's discuss offline

Reply:
Ok, if it is POT logic than I am fine with that